### DISCUSSION TITLE
Select() fails when using `order()` with `eq()`

### DISCUSSION THREAD

**Discussion by ComputelessComputer** [supabase/supabase#8876] [Questions] [answered]
I am getting a 500 error from the server when I use the `order` method after `equal`.

Does anyone know why it fails?

```javascript
const { data, error2 } = await supabase
            .from( "project_contributor" )
            .select( `
            projects(
                id,
                name,
                about,
                updated_at
            )
            ` )
            .eq( "contributor", username )
            .order( 'updated_at', { ascending: false } );
```

**Comment by GaryAustin1** [comment, marked as answer]
.eq and .order should have no issue working together.
Is there any error returned in the response?  

You use error2 in your return values, that won't work.  You need either error or error:error2.  That might give you an error reason (but is not the cause of the error).

You are ordering on updated_at.  Do you have an updated_at in project_contributor?   If not that will error (I think with something other than 500 though).

If you want to order the projects you need to add foreignTable: 'projects' with ascending:false to access that updated_at.

**Reply by ComputelessComputer** [reply]
@GaryAustin1 Thx for the reply.

> You are ordering on updated_at. Do you have an updated_at in project_contributor? If not that will error (I think with something other than 500 though).

This seems to be the issues then.  No, the `project_contributor` table does not have a  `updated_at` column. The `projects` table has it. If I were to sort the results from the query, what should I do?

(edit) I couldn't quite understand the last sentence. Could you please give me a pseudo code or example of some sort?

**Reply by GaryAustin1** [reply]
https://supabase.com/docs/reference/javascript/order#with-embedded-resources
.order( 'updated_at', { ascending: false, foreignTable:'projects' } )

**Reply by ComputelessComputer** [reply]
Thx! I'll take a look at it right now!
